docker/backend/Dockerfile

```diff
@@ -12,14 +15,18 @@ RUN npm install --omit=dev --omit=optional
 RUN npm run package
 
 FROM node:16.16.0-buster-slim
+ARG UID
+ARG GID
 
 WORKDIR /backend
 
-RUN chown 1000:1000 ./
-COPY --from=builder --chown=1000:1000 /build/package ./package/
-COPY --from=builder --chown=1000:1000 /build/mempool-config.json /build/start.sh /build/wait-for-it.sh ./
+RUN chown $UID:$GID ./
+COPY --from=builder --chown=$UID:$GID /build/package ./package/
+COPY --from=builder --chown=$UID:$GID /build/mempool-config.json /build/start.sh /build/wait-for-it.sh ./
+
 
-USER 1000
+RUN usermod -u $UID -o node && groupmod -g $GID -o node
```

Comment:
Why is this needed?

Reply:
When you look at the Dockerfile of your backend base image `node:16.16.0-buster-slim`, you will notice that the user `node` is created with default IDs.

This user has to be modified, so the process called at the Entrypoint can run under the user specified when building.

```
FROM debian:buster-slim

RUN groupadd --gid 1000 node \
  && useradd --uid 1000 --gid node --shell /bin/bash --create-home node
...

```

Reply:
I think we should enclose $UID and $GID in quotes like this `"${UID}"`

Reply:
`${UID}` and `${GID}` work fine. Tested with `docker build --build-arg UID=1001 --build-arg GID=1001 .`

What would be the benefit of changing it to `${UID}`? 

I checked [Dockerfile Reference](https://docs.docker.com/engine/reference/builder/#using-arg-variables). They didn't use this syntax either.